### ISSUE TITLE
Polymorphic conditions in abilities no longer work as of 3.0.0

### ISSUE THREAD

**Issue by frostblooded** [CanCanCommunity/cancancan#588]
### Steps to reproduce

Reproduction script: https://pastebin.com/bJjq8aFU

As far as I know, CanCanCan doesn't support polymorphic associations. That is why I used a workaround (something like https://github.com/ryanb/cancan/issues/939#issuecomment-24814252 and https://stackoverflow.com/a/25966630/3659426 combined) when I want to have conditions for an ability that will check inside the polymorphic association. This used to work correctly until I updated the gem to 3.0.1 (though the issue seems to appear as of 3.0.0).

### Expected behavior
The abilities defined in the snippet to work.

### Actual behavior
Error is thrown:

```
Error:
BugTest#test_bug:
NameError: uninitialized constant Connection::Owner
    


bin/rails test cancancan_snippet.rb:71
```

### System configuration
**Rails version**:
5.1.7

**Ruby version**:
ruby 2.4.5p335 (2018-10-18 revision 65137) [x86_64-linux]

**CanCanCan version**
3.0.1

The issue seems to be occurring because of the code in `conditions_normalizer.rb` in method `calculate_result_hash` and more specifically because of what `reflection.source_reflection_name` returns.

**Comment by eloyesp** [comment]
I'm facing the same issue and I'm working on a fix, it seems that while "normalizing" the hash it makes explicit the has_many_through associations, but that does break has_many_through_polymorphic associations, as they have no name.

I'm working on a fix.

**Comment by coorasse** [comment]
fixed in 3.0.2
